commands/proxy/proxy-set.ts

```diff
@@ -17,25 +18,42 @@ export class ProxySetCommand extends ProxyCommandBase {
 		private $prompter: IPrompter,
 		protected $analyticsService: IAnalyticsService,
 		protected $logger: ILogger,
+		protected $options: ICommonOptions,
 		protected $proxyService: IProxyService) {
 		super($analyticsService, $logger, $proxyService, proxySetCommandName);
 	}
 
 	public async execute(args: string[]): Promise<void> {
-		let hostname = args[0],
-			port = args[1],
-			username = args[2],
-			password = args[3];
+		let urlString = args[0],
+			username = args[1],
+			password = args[2];
+
+		const noUrl = !urlString;
+		if (noUrl) {
+			if (!isInteractive()) {
+				this.$errors.fail("Console is not interactive - you need to supply all command parameters.");
+			} else {
+				urlString = await this.$prompter.getString("Url", { allowEmpty: false });
+			}
+		}
 
-		const noHostName = !hostname;
+		const urlObj = parse(urlString);
+		let port = urlObj.port && +urlObj.port || HttpProtocolToPort[urlObj.protocol];
 		const noPort = !port || !this.isValidPort(port);
-
-		if (!isInteractive() && (noHostName || noPort || this.isPasswordRequired(username, password))) {
-			this.$errors.fail("Console is not interactive - you need to supply all command parameters.");
+		const authCredentials = this.getCredentialsFromAuth(urlObj.auth || "");
+		if ((username && authCredentials.username && username !== authCredentials.username) ||
+			password && authCredentials.password && password !== authCredentials.password) {
+				this.$errors.fail("The credentials you have provided in the url address mismatch those passed as command line arguments.");
 		}
-
-		if (noHostName) {
-			hostname = await this.$prompter.getString("Hostname", { allowEmpty: false });
+		username = username || authCredentials.username;
+		password = password || authCredentials.password;
+
+		if (!isInteractive()) {
+			if (noPort) {
+				this.$errors.failWithoutHelp(`The port you have specified (${port || "none"}) is not valid.`);
```

Comment:
`The port you have specified none is not valid`

This seems like a strange error message.